### ISSUE TITLE
Plugin incompatible with PhpStorm 2021.2 EAP

### ISSUE THREAD

**Issue by EdieLemoine** [Flamenco/intellivue#15]
This plugin broke PhpStorm 2021.2 EAP. I kept getting the following error until I removed IntelliVuePlugin.jar from my plugins folder.

```
Internal error. Please refer to https://jb.gg/ide/critical-startup-errors

java.util.concurrent.CompletionException: java.lang.StackOverflowError
    at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
    at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1739)
    at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:214)
    at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
    at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:196)
    at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:793)
    at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:349)
    at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
    at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
    at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
    at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
    at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
    at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
    at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
    at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
    at java.base/java.security.AccessController.doPrivileged(Native Method)
    at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
    at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:406)
    at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
    at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
    at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
    at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
    at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.StackOverflowError
    at com.intellij.util.lang.ZipResourceFile.findClass(ZipResourceFile.java:113)
    at com.intellij.util.lang.JarLoader.findClass(JarLoader.java:55)
    at com.intellij.util.lang.ClassPath.findClassInLoader(ClassPath.java:217)
    at com.intellij.util.lang.ClassPath.findClass(ClassPath.java:176)
    at com.intellij.ide.plugins.cl.PluginClassLoader.loadClassInsideSelf(PluginClassLoader.java:370)
    at com.intellij.ide.plugins.cl.PluginClassLoader.tryLoadingClass(PluginClassLoader.java:265)
    at com.intellij.serviceContainer.ComponentManagerImplKt.doLoadClass(ComponentManagerImpl.kt:1455)
    at com.intellij.serviceContainer.ComponentManagerImplKt.access$doLoadClass(ComponentManagerImpl.kt:1)
    at com.intellij.serviceContainer.ComponentManagerImpl.loadClass(ComponentManagerImpl.kt:807)
    at com.intellij.serviceContainer.BaseComponentAdapter.getImplementationClass(BaseComponentAdapter.kt:38)
    at com.intellij.serviceContainer.BaseComponentAdapter.getComponentImplementation(BaseComponentAdapter.kt:31)
    at com.intellij.serviceContainer.MyComponentAdapter.toString(MyComponentAdapter.kt:65)
    at com.intellij.serviceContainer.BaseComponentAdapter.getImplementationClass(BaseComponentAdapter.kt:41)
    at com.intellij.serviceContainer.BaseComponentAdapter.getComponentImplementation(BaseComponentAdapter.kt:31)
    at com.intellij.serviceContainer.MyComponentAdapter.toString(MyComponentAdapter.kt:65)
    at com.intellij.serviceContainer.BaseComponentAdapter.getImplementationClass(BaseComponentAdapter.kt:41)
    at com.intellij.serviceContainer.BaseComponentAdapter.getComponentImplementation(BaseComponentAdapter.kt:31)
    at com.intellij.serviceContainer.MyComponentAdapter.toString(MyComponentAdapter.kt:65)
    at com.intellij.serviceContainer.BaseComponentAdapter.getImplementationClass(BaseComponentAdapter.kt:41)
    at com.intellij.serviceContainer.BaseComponentAdapter.getComponentImplementation(BaseComponentAdapter.kt:31)
   ....
  

-----
Your JRE: 11.0.11+9-b1481.1 x86_64 (JetBrains s.r.o.)
~/Library/Application Support/JetBrains/Toolbox/apps/PhpStorm/ch-0/212.3724.24/PhpStorm 2021.2 EAP.app/Contents/jbr/Contents/Home
```

I tried installing the plugin again from the Plugins browser in PhpStorm itself, restarted and got the same error.

**Comment by Flamenco** [comment]
Thanks for the report.  So it worked until you upgraded from 2021.1 EAP to 2021.2 EAP, correct?

**Comment by Flamenco** [comment]
I upgraded to 2021.2 EAP and confirm this error occurs, and is resolved by removing the plugin.  However, my IDE will not launch even without the plugin installed.

I have a feeling this might be a broken EAP in general, so I'm going to give JetBrains a few days to get their act together.

**Comment by elie-g** [comment]
Same bug on WebStorm

**Comment by Flamenco** [comment]
An update... We are still waiting for a stable  EAP to be released before working on this.

**Comment by EdieLemoine** [comment]
I get that you're not going to fix this immediately, but other plugins that were outdated didn't cause this problem because they have proper version constraints. This is the first time a plugin has caused this type of error for me after updating. 

Here's an example of a plugin that got disabled automatically:
<img width="363" alt="Screenshot 2021-06-08 at 16 12 06" src="https://user-images.githubusercontent.com/3886637/121200768-48519600-c874-11eb-8e92-fb90105858b5.png">

**Comment by Flamenco** [comment]
@EdieLemoine Yes, first time for us as well.  JetBrains it supposed to sandbox the plugins so a hard crash like this does not happen.

**Comment by Flamenco** [comment]
Jetbrains release a new EAP today, so we should get to the bottom of the issue by days end.

**Comment by Flamenco** [comment]
We submitted an update today. Version 1.1.4.  This update adds no features.  It specifies in metadata that the plugin will not work with IJ 2021.2 and higher.

We also updated the compatibility range for version 1.1.3 on JetBrains plugin portal.

A new version will be released after the EAP has become stable.  Until then we will keep this issue open.

**Comment by Flamenco** [comment]
@EdieLemoine I was able to set the versions on the website for all previous releases as well, so they all now display the message you posted.  The website version setting overrides the version in the plugin even.

I am still having major problems trying to debug plugins with 2021.2 EAP.  Runs really slow and hangs.  Not usable for me ATM.

**Comment by Flamenco** [comment]
<img width="526" alt="Screen Shot 2021-06-22 at 10 21 19 AM" src="https://user-images.githubusercontent.com/819940/122941775-b2be0800-d343-11eb-82e8-d0ad712f45ea.png">

**Comment by EdieLemoine** [comment]
@Flamenco that's great, at least it won't break IDEs anymore!

I'll manage without this plugin for a while, it's a sacrifice I have to make either way by being too impatient to wait for the bigger releases. :)

**Comment by Flamenco** [comment]
Fixed in release 1.1.6.  It may take a day or two for Jetbrains to publish, so I added the JAR to the release page for the impatient.

https://github.com/Flamenco/intellivue/releases
